cmd/zc_enumerator.go

```diff
@@ -72,6 +82,112 @@ func newStoredObject(name string, relativePath string, lmt time.Time, size int64
 // pass each storedObject to the given objectProcessor if it passes all the filters
 type resourceTraverser interface {
 	traverse(processor objectProcessor, filters []objectFilter) error
+	isDirectory() bool
+}
+
+// source, location, recursive, and incrementEnumerationCounter are always required.
+// ctx, pipeline are only required for remote resources.
+// followSymlinks is only required for local resources (defaults to false)
+func initResourceTraverser(source string, location common.Location, ctx *context.Context, credential *common.CredentialInfo, followSymlinks *bool, listOfFilesLocation *string, recursive bool, incrementEnumerationCounter func()) (resourceTraverser, error) {
+	var output resourceTraverser
+	var p *pipeline.Pipeline
+
+	if location == common.ELocation.Local() {
+		source = cleanLocalPath(source)
+	}
+
+	if ctx != nil && credential != nil {
+		tmppipe, err := initPipeline(*ctx, location, *credential)
+
+		if err != nil {
+			return nil, err
+		}
+
+		p = &tmppipe
+	}
+
+	if listOfFilesLocation != nil && *listOfFilesLocation != "" {
+		splitsrc := strings.Split(source, "?")
+		sas := ""
+
+		if len(splitsrc) > 1 {
+			sas = splitsrc[1]
+		}
+
+		f, err := os.Open(*listOfFilesLocation)
+
+		if err != nil {
+			return nil, err
+		}
+
+		output = newListTraverser(splitsrc[0], sas, location, credential, ctx, recursive, f)
+		return output, nil
+	}
+
+	switch location {
+	case common.ELocation.Local():
+		toFollow := false
+		if followSymlinks != nil {
+			toFollow = *followSymlinks
+		}
+
+		if strings.Index(source, "*") != -1 {
+			basePath := trimWildcards(source)
+			matches, err := filepath.Glob(source)
+
+			if err != nil {
+				return nil, fmt.Errorf("failed to glob: %s", err)
+			}
+
+			// A channel really would be preferable for this kind of behaviour but whatever.
```

Comment:
bytes.Buffer achieves the functionality I was aiming for.

Unless you'd like me to adjust it so that the list traverser reads until a channel closes. The upside of this is that we could spin up a routine to read a file/glob instead of letting the traverser read line-by-line, so that these would occur side-by-side.

Reply:
Hmm a channel would indeed be much better. If you are ok with it I'd suggest to adopt the channel approach.